src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -306,14 +306,19 @@ public virtual void Write(ulong value)
         //
         public virtual unsafe void Write(float value)
         {
-            uint TmpValue = *(uint*)&value;
-            _buffer[0] = (byte)TmpValue;
-            _buffer[1] = (byte)(TmpValue >> 8);
-            _buffer[2] = (byte)(TmpValue >> 16);
-            _buffer[3] = (byte)(TmpValue >> 24);
+            BinaryPrimitives.WriteSingleLittleEndian(_buffer, value);
```

Comment:
since you have changed it for `single` overload, should we do the same for `int`, `uint`, `short`, `ushort` and other methods that don't use `BinaryPrimitives`? (just for consistency)

https://github.com/dotnet/runtime/blob/88cf0ed1dbab5c4529b64dca04a883e7178af285/src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs#L242-L283

Reply:
This was explored in https://github.com/dotnet/runtime/pull/37582#issue-430885578 . Using the Span-based helpers for <8 byte types is not profitable (converting array to Span is cheap, but it is not free).

Reply:
> Using the Span-based helpers for <8 byte types is not profitable 

@jkotas  thanks for another valuable comment! Does it mean that the `single` change should be reverted and the new `Write(Half value)` method should not be using `BinaryPrimitives` as well?

Reply:
I would think so, unless we have data that suggest otherwise.